gitdiff/binary.go

```diff
@@ -56,7 +59,9 @@ func (p *parser) ParseBinaryMarker() (isBinary bool, hasData bool, err error) {
 	case "Binary files differ\n":
 	case "Files differ\n":
 	default:
-		return false, false, nil
+		if !binaryRegexp.MatchString(p.Line(0)) {
```

Comment:
I'm wondering if we should optimize this to `strings.Contains(p.Line(0), "Binary files")` instead of using a regex. Gitleaks really shouldn't be inspecting any binary data.

Or perhaps maybe `func (p *parser) ParseBinaryFragments(f *File) (n int, err error) {` should always return `false, false, nil`.

WDYT @k0ral?

@weineran might you have thoughts on this?

Reply:
> I'm wondering if we should optimize this to `strings.Contains(p.Line(0), "Binary files")`

This sounds quite lax to me, how confident are we that the string "Binary files" can't appear anywhere else in a diff ?
Also, I realize now that `MatchString` is actually a "contains" rather than a "exactly matches", so for optimization sake, I guess we should enclose the pattern with `^$` to allow an early return.

> Or perhaps maybe `func (p *parser) ParseBinaryFragments(f *File) (n int, err error) {` should always `return false, false, nil`.

Not sure I understand your proposal, as types don't match in your sentence :) . Did you mean:
- `ParseBinaryMarker` should always return `false, false, nil` ?
- or `ParseBinaryFragments` should always return `0, nil` ?

Reply:
> ParseBinaryMarker should always return false, false, nil ?

Whoops, this is what I meant. Good catch 😉

> I guess we should enclose the pattern with ^$ to allow an early return, shouldn't we ?

That makes sense to me

Reply:
If `parseBinaryMarker` always returned `false, false, nil`, then:
- [`f.IsBinary`](https://github.com/gitleaks/go-gitdiff/pull/1/files#diff-17e95fb6f9c8f7c810a17f1f3cc94b776e2a67eb25a46f6cabba1ae8fa5fcd6aR22) would always be `false`
- ... which would essentially treat all files as non-binary
- ... which would be incorrect, right ? :)

Reply:
You're catching me before having some caffeine in my system! ☕ 

> If parseBinaryMarker always returned false, false, nil, then:
... which would be incorrect, right ? :)

You are right, that would be incorrect. What we are aiming for is if a binary file has been detected, treat that file as if it has no data. In other words, gitleaks should never scan binary files (for now). However, that logic might be better for another PR as this PR is a `fix`.

I think once `I guess we should enclose the pattern with ^$ to allow an early return, shouldn't we ?` has been added these changes look good to me.

Reply:
Pull-request has been updated :) .